delphi-checks/src/main/java/au/com/integradev/delphi/utils/RoutineUtils.java

```diff
@@ -76,25 +77,32 @@ private static Stream<Type> concreteParentTypesStream(Type type) {
 
   private static boolean isOverriddenMethod(
       RoutineNameDeclaration parent, RoutineNameDeclaration child) {
-    if (parent.getName().equalsIgnoreCase(child.getName())
-        && parent.getParameters().equals(child.getParameters())) {
-      if (parent.isClassInvocable()) {
-        if (parent.getRoutineKind() == RoutineKind.CONSTRUCTOR
-            || parent.getRoutineKind() == RoutineKind.DESTRUCTOR) {
-          // An instance constructor or destructor cannot inherit from a class constructor or
-          // destructor
-          return child.isClassInvocable();
-        } else {
-          // Any other type of invocable can inherit from a class invocable
-          return true;
-        }
-      } else {
-        // A class invocable cannot inherit from an instance invocable
-        return !child.isClassInvocable();
-      }
-    } else {
+    if (!(parent.getName().equalsIgnoreCase(child.getName())
+        && parent.getParametersCount() == child.getParametersCount())) {
+      return false;
+    }
+
+    if (!IntStream.range(0, parent.getParametersCount())
+        .allMatch(
+            param ->
+                parent.getParameter(param).getType().is(child.getParameter(param).getType()))) {
       return false;
     }
+
+    if (!parent.isClassInvocable()) {
+      // A class invocable cannot inherit from an instance invocable
+      return !child.isClassInvocable();
+    }
+
+    if (parent.getRoutineKind() == RoutineKind.CONSTRUCTOR
+        || parent.getRoutineKind() == RoutineKind.DESTRUCTOR) {
+      // An instance constructor or destructor cannot inherit from a class constructor or
+      // destructor
+      return child.isClassInvocable();
+    }
+
+    // Any other type of invocable can inherit from a class invocable
+    return true;
```

Comment:
As far as I can tell, this change isn't relevant to improving `RedundantInheritedCheck`.

Reply:
This change alters the way parameters are equated, i.e., parameter lists are now equal if the parameter names are different.